### ISSUE TITLE
Multi-GPU support for GNNs

### ISSUE THREAD

**Issue by francois-drielsma** [DeepLearnPhysics/lartpc_mlreco3d#32]
GNN edge models seem to run into problems when running on multiple GPUs:
```
RuntimeError: arguments are located on different GPUs at /pytorch/aten/src/THC/generic/THCTensorMathBlas.cu:255
```

**Comment by francois-drielsma** [comment]
Issues to be resolved for multi-GPU support have been identified as follows:
1. https://github.com/rusty1s/pytorch_geometric/issues/535: must restructure `MetaLayer` objects so that the predictors passed to it are classes that explicitly inherit from `torch.nn.Module` (e.g. https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.meta.MetaLayer);
2. The forward functions of the GNNs must return arrays of tensors instead of dictionaries to conform with what the `gather` function of object `DataParallel` in `data_parallel.py` expects as an input;
3. The loss functions of the GNNs must loop over the different GPU outputs and concatenate the losses and accuracies in a sensible way.

I will start to implement those changes.

**Comment by francois-drielsma** [comment]
Fixed in PR #36.
